src/raster/dataset.rs

```diff
@@ -49,11 +48,22 @@ impl Dataset {
 
     pub unsafe fn _c_ptr(&self) -> *const c_void {
         return self.c_dataset;
-    }
-
-    pub fn size(&self) -> (isize, isize) {
-        let size_x = unsafe { gdal::GDALGetRasterXSize(self.c_dataset) } as isize;
-        let size_y = unsafe { gdal::GDALGetRasterYSize(self.c_dataset) } as isize;
+    }
+
+
+    pub fn get_rasterband<'a>(&'a self, band_index: isize) -> Option<RasterBand<'a>> {

```

Comment:
It's Rust convention to not have `get_` prefixes on getters, but it's up to you.

https://doc.rust-lang.org/style/style/naming/README.html#gettersetter-methods-rfc-344